test/integration_test.py

```diff
@@ -405,6 +410,54 @@ def _get_entities_by_project(self, entity_type, project_short_name):
             params['search_after_uid'] = pagination['search_after_uid']
         return entities
 
+    @unittest.skipIf(config.deployment_stage not in ('sandbox', 'nadove'),
+                     'Not polluting portal db since not in sandbox environment')
+    def test_concurrent_portal_db_crud(self):
+        """
+        Use multithreading to simulate multiple users simultaneously modifying
+        the portals database.
+        """
+
+        # Currently takes about 50 seconds and creates a 25 kb db file.
+        n_threads = 10
+        n_ops = 10
+        portal_service = PortalService()
+
+        entry_format = 'thread={};op={}'
+
+        def run(thread_count):
+            for op_count in range(n_ops):
+                mock_entry = cast(JSON, {
+                    "portal_id": "foo",
+                    "integrations": [
+                        {
+                            "integration_id": "bar",
+                            "entity_type": "project",
+                            "integration_type": "get",
+                            "entity_ids": ["baz"]
+                        }
+                    ],
+                    "mock-count": entry_format.format(thread_count, op_count)
+                })
+                portal_service._crud(lambda db: list(db) + [mock_entry])
+
+        old_db = portal_service.read()
+
+        with ThreadPoolExecutor(max_workers=n_threads) as executor:
+            for i in range(n_threads):
```

Comment:
I'm not sure how quickly TPE allocates new threads. It might make sense to schedule an order of magnitude more tasks than the pool can run in parallel.

```suggestion
            for i in range(n_threads * 10):
```